src/clp_ffi_js/ir/StreamReader.hpp

```diff
@@ -50,42 +53,54 @@ class StreamReader {
     [[nodiscard]] auto get_num_events_buffered() const -> size_t;
 
     /**
-     * Deserializes and buffers log events in the range `[beginIdx, endIdx)`. After the stream has
-     * been exhausted, it will be deallocated.
+     * @return The filtered log events map.
+     */
+    [[nodiscard]] auto get_filtered_log_event_map() const -> FilteredLogEventMapType;
+
+    /**
+     * Filters log events and generates `m_filtered_log_event_map`. If `logLevelFilter` is `null`,
+     * `m_filtered_log_event_map` will be set to `std::nullopt`.
```

Comment:
I feel the docs are kinda specific to the implementation. How about
```suggestion
     * Generates a filtered collection from all log events.
```
Below we can mention that the collection is cleared when `logLevelFilter` is null.

Alternatively, we can mark this as a private method (so the docs can be implementation specific) and add the `setLogLevelFilter` method here. What do you think?